File: Homework1_Derezovskiy/tetrahedron.cpp
//------------------------------------------------------------------------------
// tetrahedron.cpp - содержит функции обработки тетраэдра
//------------------------------------------------------------------------------

#include "math.h"
#include "tetrahedron.h"

//------------------------------------------------------------------------------
// Ввод параметров тетраэдра из потока
void In(tetrahedron &t, ifstream &ifst) {
    ifst >> t.a >> t.materialDensity;
}

// Случайный ввод параметров тетраэдра
void InRnd(tetrahedron &t) {
    t.a = Random();
    t.materialDensity = Random();
}

//------------------------------------------------------------------------------
// Вывод параметров тетраэдра в поток
void Out(tetrahedron &t, ofstream &ofst) {
    ofst << "It is Tetrahedron: a = " << t.a << ". Material density = " << t.materialDensity
         << ". Square = " << Square(t) << "\n";
}

//------------------------------------------------------------------------------
// Вычисление площади поверхности тетраэдра
double Square(tetrahedron &t) {
    return double(t.a * t.a * sqrt(3));
}

File: Homework3_Derezovskiy/src/tetrahedron.py
import math
import random

class Tetrahedron:
    def __init__(self):
        self.a = 0
        self.materialDensity = 0

    def In(self, strArray):
        self.a = int(strArray[1])
        self.materialDensity = int(strArray[2])

    def InRnd(self):
        self.a = random.randint(1, 20)
        self.materialDensity = random.randint(1, 20)

    def Write(self, ostream):
        ostream.write("Tetrahedron: a = {}, materialDensity = {}, Square = {}".
                      format(self.a, self.materialDensity, self.Square()))
        pass

    def Square(self):
        return float(self.a * self.a * math.sqrt(3));
        pass
